src/content/docs/en/guides/cms/builderio.mdx

```diff
@@ -3,19 +3,507 @@ title: Builder.io & Astro
 description: Add content to your Astro project using Builder.io’s visual CMS
 type: cms
 service: Builder.io
-stub: true
+stub: false
 i18nReady: false
 ---
 import PackageManagerTabs from '~/components/tabs/PackageManagerTabs.astro'
+import FileTree from '~/components/FileTree.astro';
 
+[Builder.io](https://www.builder.io/) is a visual CMS that supports drag and drop content editing for building web sites.
 
+At the end of this recipe you'll be able to create blog posts in the Builder app and have your Astro site update with this new content automatically using a webhook.
+
+To do this, you'll use Builder's [Content API](https://www.builder.io/c/docs/content-api) and [Qwik API](https://www.builder.io/c/docs/qwik-api) to connect your Builder space to Astro with zero client-side JavaScript.
+
+:::tip[Official resource]
+If you are using a JavaScript framework (e.g. Svelte, Vue, or React) in your Astro project you can use one of Builder's integrations as an alternative to making raw fetch calls through the REST API.
+:::
+
+## Prerequisites
+
+To get started, you will need to have the following:
+
+* **A Builder account and space** - If you don't have an account yet, [sign up for free](https://www.builder.io/) and create a new space. If you already have a space with Builder, feel free to use it, but you will need to modify the code to match the model name and custom data fields.
+* **A Builder API key** - This public key will be used to fetch your content from Builder. [Read Builder's guide on how your key](https://www.builder.io/c/docs/using-your-api-key#finding-your-public-api-key).
+
+## Setting up credentials
+
+To add your Builder API key to Astro, create a `.env` file in the root of your project with the following variable:
+
+```ini title=".env"
+BUILDER_API_PUBLIC_KEY=YOUR_API_KEY
+```
+
+Now, you should be able to use this API key in your project.
+
+:::note
+At the time of writing, [this key is public](https://www.builder.io/c/docs/using-your-api-key#prerequisites), so you don't have to worry about hiding or encrypting it.
+:::
+
+If you would like to have IntelliSense for your environment variables, you can create a `env.d.ts` file in the `src/` directory and configure `ImportMetaEnv` like this:
+
+```ts title="src/env.d.ts"
+interface ImportMetaEnv {
+  readonly BUILDER_API_PUBLIC_KEY: string;
+}
+```
+
+Your project should now include these new files:
```

Comment:
```suggestion
Your project should now include these files:
```
Just in case they weren't new!

Reply:
Also marked as resolved! So, we need to decide what this means. Is this your way of saying, "OK! Go ahead and change this!" or "I see what you wrote, and I'm ignoring it."  :sweat_smile:

Reply:
hahaha, nonono I wasn't ignoring anything here. I wouldn't miss any of this :D. Resolve means go ahead. Let me know if this was the way to go or if You expected like a verbal confirmation or something.

Reply:
OK, thanks! The normal way to indicate you're happy is to click "Commit suggestion". "Resolved" means, I consider the current version of the page acceptable... we're done here. "I said good day!"  lol

I will revisit these "resolved" comments in light of that.  :smile:

Reply:
ooooh. learning! 😁 

yes, so whenever i *resolved* i meant to *commit the suggestion*. will go back and do some amends :)